### DISCUSSION TITLE
Connect using SRV records

### DISCUSSION THREAD

**Discussion by luddd3** [nodejs/undici#2067] [Q&A] [answered]
I am using a client with a long-lived connection to a remote server. The server DNS use SRV records instead of static IP addresses. What is the best way, using undici, to handle reconnects etc? Do I need to resolve the SRV records before every fetch or is there a more efficient way?

**Comment by metcoder95** [comment]
Hey @luddd3, did you find a solution to this?

**Reply by luddd3** [reply]
Unfortunately no, I am in no hurry either, but if there is no good built in solution I would look for/implement some kind of hook that would run pre-connect.
I would of course appreciate if someone with better knowledge of the code base could implement it or point in the right direction.

**Reply by metcoder95** [reply, marked as answer]
I would say that I've not faced this problem yet, although you can customize the behaviour and the way to use sockets while connecting to servers by passing a custom `connect` function at the Client level, see: https://undici.nodejs.org/#/docs/api/Client?id=example-custom-connector

This can help you run your own custom resolutions everything a socket connection is made
